src/Type/Php/MethodExistsTypeSpecifyingExtension.php

```diff
@@ -45,18 +48,49 @@ public function specifyTypes(
 	): SpecifiedTypes
 	{
 		$methodNameType = $scope->getType($node->args[1]->value);
-		if (!$methodNameType instanceof ConstantStringType) {
+		$methodNames = $this->getMethodNames($methodNameType);
+		if (count($methodNames) === 0) {
 			return new SpecifiedTypes([], []);
 		}
 
 		return $this->typeSpecifier->create(
 			$node->args[0]->value,
 			new IntersectionType([
 				new ObjectWithoutClassType(),
-				new HasMethodType($methodNameType->getValue()),
+				TypeCombinator::union(...array_map(
```

Comment:
1) This change has no test
2) The whole `getMethodNames` method can be replaced with calling `TypeUtils::getConstantStrings()`